crates/router/src/connector/nuvei/transformers.rs

```diff
@@ -487,167 +540,187 @@ fn get_card_info<F>(
-                payment_option: PaymentOption {
-                    card: Some(Card {
-                        card_number: Some(card.card_number),
-                        card_holder_name: Some(card.card_holder_name),
-                        expiration_month: Some(card.card_exp_month),
-                        expiration_year: Some(card.card_exp_year),
-                        cvv: Some(card.card_cvc),
-                        ..Default::default()
-                    }),
-                    ..Default::default()
-                },
+                payment_option: PaymentOption::from(NuveiCardDetails {
+                    card,
+                    three_d: None,
+                }),
                 ..Default::default()
             }),
             _ => Err(errors::ConnectorError::NotImplemented(
                 "Payment methods".to_string(),
             )),
         }?;
-        Ok(Self {
-            merchant_id: merchant_id.clone(),
-            merchant_site_id: merchant_site_id.clone(),
-            client_request_id: client_request_id.clone(),
+        let request = Self::try_from(NuveiPaymentRequestData {
             amount: item.request.amount.clone().to_string(),
             currency: item.request.currency.clone().to_string(),
-            transaction_type: item
-                .request
-                .capture_method
-                .map(TransactionType::from)
-                .unwrap_or_default(),
-            time_stamp: time_stamp.clone(),
-            session_token,
-            checksum: encode_payload(vec![
-                merchant_id,
-                merchant_site_id,
-                client_request_id,
-                item.request.amount.to_string(),
-                item.request.currency.to_string(),
-                time_stamp,
-                merchant_secret,
-            ])?,
-            ..request_data
+            connector_auth_type: item.connector_auth_type.clone(),
+            client_request_id: item.attempt_id.clone(),
+            session_token: data.1,
+            capture_method: item.request.capture_method,
+            ..Default::default()
+        })?;
+        Ok(Self {
+            related_transaction_id: request_data.related_transaction_id,
+            payment_option: request_data.payment_option,
+            ..request
         })
     }
 }
 
-impl TryFrom<&types::PaymentsCaptureRouterData> for NuveiPaymentFlowRequest {
+impl TryFrom<NuveiPaymentRequestData> for NuveiPaymentsRequest {
     type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(item: &types::PaymentsCaptureRouterData) -> Result<Self, Self::Error> {
-        let connector_meta: NuveiAuthType = NuveiAuthType::try_from(&item.connector_auth_type)?;
+    fn try_from(request: NuveiPaymentRequestData) -> Result<Self, Self::Error> {
+        let session_token = request.session_token;
+        if session_token.is_empty() {
+            return Err(errors::ConnectorError::MissingRequiredField {
```

Comment:
same here, we can avoid the usage of return keyword. `fp_utils::when` should do the job. Also the error should be `FailedToObtainAuthType`